### ISSUE TITLE
Occasionally empty datasets on simple reductions

### ISSUE THREAD

**Issue by djspiewak** [quasar-analytics/quasar#2559]
The queries which do this generally return very fast, but with no results.  Exemplars:

- `groupByArray.test`
- `locationCount.test`
- `statesByShortestFirstCity.test`

**Comment by djspiewak** [comment]
`groupByArray.test` is now fixed (as of 20.13.14).  `statesByShortestFirstCity.test` is an issue with `MIN` (and the fact that it is being applied to strings).  `locationCount.test` is still mysterious.

**Comment by djspiewak** [comment]
Here's some minimization:

```sql
select count(*) as counter, PositionHeader.PositionLocation.LocationCity from jobs_jobinfo group by PositionHeader.PositionLocation.LocationCity
```

This returns correct results instantaneously, with very simple qscript (`Reduce` root node).

```sql
select count(*) as counter, PositionHeader.PositionLocation.LocationCity as location from jobs_jobinfo group by location
```

This returns the empty result set after a delay, with very complex qscript and a root join node.  I'm pretty sure this is qscript not detecting the common sources again, and may be fixed by the work @alissapajer is doing right now.  I'll leave this one open, but we should verify after #2821 is fixed.

**Comment by djspiewak** [comment]
Oh, also removing the **Mimir** label, since it's actually an issue in qscript that is causing this.

**Comment by zeeiyerWork** [comment]
Adding the Qscript label based on the above comment.

**Comment by djspiewak** [comment]
Upon investigation with Alissa, this isn't even a qscript issue!  It's actually an issue in the SQL^2 parser!  The `as location` in the `SELECT` clause is being parsed as a projection, rather than a map creation.  This bug is literally 3 years old.

**Comment by alissapajer** [comment]
The bug is present in a query as simple as
```sql
select city as City from olympics group by City
```

**Comment by alissapajer** [comment]
^^ This is not a bug because that query is grouping on an alias, which is not currently allowed by quasar. Will reconsider the original query and report back.

**Comment by alissapajer** [comment]
As of 45c2d4b87050fabb21bb5f078427b975bc4b3eb5:

`groupByArray.test` is passing for all connectors, and is not trying to group by an alias.

`statesByShortestFirstCity.test` is passing for all connectors except for CB and mongo 3.0 and less.

`locationCount.test` is failing for all connectors. It is also attempting to group by a field alias in the test query! I will update this test to group by the actual field and investigate from there. This failing test is already tracked in https://github.com/quasar-analytics/quasar/issues/2494.

I am closing this issue because there is already an issue tracking the only remaining reported problem.
